### DISCUSSION TITLE
Volume in streams-generator-a2dp.ino

### DISCUSSION THREAD

**Discussion by monojk** [pschatzmann/arduino-audio-tools#142] [Q&A] [answered]
I ran the code from the example streams-generator-a2dp.ino on an ESP32 sending to a "Teufel BAMSTER XS". 
I get a tone, but it is too loud even with out.setVolume(1). Also the volume is increasing and cannot be controlled by the volume buttons on the BT speaker. Any Idea?

**Comment by pschatzmann** [comment, marked as answer]
I think the value needs to be between 0 and 1.0 so 1 is 100%.
I suggest that you try with 0.1

**Comment by monojk** [comment]
Thanks, that worked.
Another problem came up when trying to use the reconnect feature:
SP32_Audio:38:7: error: 'class audio_tools::A2DPStream' has no member named 'auto_reconnect'

   out.auto_reconnect = true;  // if this is use we just quickly connect to the last device ignoring cfg.name

**Comment by pschatzmann** [comment]
auto_reconnect  is an attribute of A2DPConfig: I adjusted the example.

**Comment by monojk** [comment]
Probably the statement should be cfgA2DP.auto_reconnect = true;  instead id cfg1 ... false?

**Comment by pschatzmann** [comment]
absolutely yes!
